### PR TITLE
Validation

### PR DESCRIPTION
**Pull request by re-pesk**

### CODE REVIEW

File: routes/documents.js [hunk 1]
```diff
@@ -1,16 +1,18 @@
-const express = require('express')
-const router = express.Router()
-const Document = require('../models/document')
+const express = require('express');
+const router = express.Router();
+const Document = require('../models/document');
+const DocumentBodySchema = require('./validators/validators').DocumentBodySchema;
```
Comment (ralekna) [inline routes/documents.js]: You can use object destructuring here:
`const { DocumentBodySchema } = require('./validators/validators');`

File: routes/users.js [hunk 2]
```diff
@@ -1,48 +1,59 @@
...
+// const cors = require('cors')
+const router = app.Router();
 const User = require('../models/user')
+const UserBodySchema = require('./validators/validators').UserBodySchema;
+const validator = require('express-joi-validator');
 
 /* GET users listing. */
 router.get('/', function(req, res, next) {
   User.find({})
+    console.log(records)
```
Comment (ralekna) [inline routes/users.js]: Will you need this?

File: routes/users.js [hunk 3]
```diff
@@ -1,48 +1,59 @@
...
     .then(records => {
       res.send(records);
     })
+    console.log(next)
```
Comment (ralekna) [inline routes/users.js]: Will you need this?

File: routes/users.js [hunk 4]
```diff
@@ -1,48 +1,59 @@
...
+router.post('/', validator(UserBodySchema), (req, res, next) => {
+    User.create(req.body)
+      .then(added => {
+        res.send(added);
+      })
+      .catch(next);
+  }
+);
+
+// router.get('/id::usrId;abc::abc', (req, res, next) => {
```
Comment (ralekna) [inline routes/users.js]: Do not leave commented out code. 99% of the time commented code will never be used and is a visual trash.

File: routes/validators/validators.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    owner: ObjectIdValidator,
+    authors: Joi.array().items(ObjectIdValidator),
+    title: Joi.string(),
+    content: Joi.string()
+};
+
+const DocumentArrayValidator = Joi.array().items(DocumentValidator);
+
+const DocumentBodySchema = {
+    body: Joi.alternatives().try(DocumentValidator, DocumentArrayValidator)
```
Comment (ralekna) [inline routes/validators/validators.js]: In what cases there will be an array Documents?

File: routes/validators/validators.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+const UserArrayValidator = Joi.array().items(UserValidator)
+
+const UserBodySchema = {
+    body: Joi.alternatives().try(UserValidator, UserArrayValidator)
+};
+  
+module.exports = {
+    UserBodySchema, 
+    DocumentBodySchema
+}
```
Comment (ralekna) [inline routes/validators/validators.js]: Please leave an empty line at the end of the file. You IDE may have an option to put it automacally.

File: routes/validators/validators.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    firstName: NonEmptyString,
+    lastName: NonEmptyString,
+    email: EmailString,
+    password: PasswordString
+}
+
+const UserArrayValidator = Joi.array().items(UserValidator)
+
+const UserBodySchema = {
+    body: Joi.alternatives().try(UserValidator, UserArrayValidator)
```
Comment (ralekna) [inline routes/validators/validators.js]: Same as for DocumentsBodySchema

**Review by ralekna — CHANGES_REQUESTED** [review]

File: routes/validators/validators.js (hunk 5 above)
Comment (re-pesk) [reply routes/validators/validators.js]: Without it, /documents accepts all array without validation.

File: routes/validators/validators.js (hunk 7 above)
Comment (re-pesk) [reply routes/validators/validators.js]: Same as in case of /documents.

File: routes/validators/validators.js (hunk 5 above)
Comment (re-pesk) [reply routes/validators/validators.js]: Removed

File: routes/validators/validators.js (hunk 7 above)
Comment (re-pesk) [reply routes/validators/validators.js]: Removed

**Review by ralekna — APPROVED** [review]

**Pull request merged**